src/main/anovos/data_analyzer/quality_checker.py

```diff
@@ -30,40 +44,38 @@ def duplicate_detection(spark, idf, list_of_cols='all', drop_cols=[], treatment=
               Metric Dataframe is of schema [metric, value] and contains metrics - number of rows, number of unique rows, 
               number of duplicate rows and percentage of duplicate rows in total.
     """
-    if list_of_cols == 'all':
-        num_cols, cat_cols, other_cols = attributeType_segregation(idf)
-        list_of_cols = num_cols + cat_cols
-    if isinstance(list_of_cols, str):
-        list_of_cols = [x.strip() for x in list_of_cols.split('|')]
-    if isinstance(drop_cols, str):
-        drop_cols = [x.strip() for x in drop_cols.split('|')]
+    if columns == 'all':
+        num_cols, cat_cols, other_cols = attributeType_segregation(input_df)
+        columns = num_cols + cat_cols
 
-    list_of_cols = list(set([e for e in list_of_cols if e not in drop_cols]))
+    columns = _parse_columns(columns)
 
-    if any(x not in idf.columns for x in list_of_cols) | (len(list_of_cols) == 0):
-        raise TypeError('Invalid input for Column(s)')
-    if str(treatment).lower() == 'true':
-        treatment = True
-    elif str(treatment).lower() == 'false':
-        treatment = False
-    else:
-        raise TypeError('Non-Boolean input for treatment')
+    drop_columns = drop_columns or []
+    drop_columns = _parse_columns(drop_columns)
+    if drop_columns:
+        columns = [col for col in columns if col not in drop_columns]
+
+    if not columns:
+        raise ValueError("No columns to work on.")
 
-    odf_tmp = idf.drop_duplicates(subset=list_of_cols)
-    odf = odf_tmp if treatment else idf
```

Comment:
Instead of introducing a "temporary" variable, we can decide what to return at the end of the function